tsdb/index/index.go

```diff
@@ -1160,6 +1160,9 @@ func newReader(b ByteSlice, c io.Closer) (*Reader, error) {
 		}
 		// Trim any extra space in the slices.
 		for k, v := range r.postings {
+			if len(v) == cap(v) {
```

Comment:
What are the chances that the number of postings is exactly a power of 2? This seems like over-optimisation.

Reply:
Although the probability is not that high, I still think this is a valid change to reduce some useless slice copy.

Reply:
It might on very rare occasion save 1ms on a process that takes minutes, it's not worth it.